app/models/miq_request_workflow.rb

```diff
@@ -33,6 +33,13 @@ def self.all_encrypted_options_fields
     descendants.flat_map(&:encrypted_options_fields).uniq
   end
 
+  def self.possibly_update_user(data, user)
+    return user if data[:user_name].blank?
+    new_user = User.lookup_by_identity(data[:user_name])
+    _log.warn "Web-service requester changed to <#{new_user.userid}>"
```

Comment:
@syncrou 
new_user could be nil, should we have an error message to indicate that the user could not be found.